parsl/monitoring/db_manager.py

```diff
@@ -365,7 +365,7 @@ def _migrate_logs_to_internal(self, logs_queue, queue_tag, kill_event):
             self.logger.debug("""Checking STOP conditions for {} threads: {}, {}"""
                               .format(queue_tag, kill_event.is_set(), logs_queue.qsize() != 0))
             try:
-                x, addr = logs_queue.get(block=False)
+                x, addr = logs_queue.get()
```

Comment:
If we block here, then monitoring will not respond to kill_event being set. We should block for some wait_time.